lib/ap/plan9/tmpfile.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file is part of the UCB release of Plan 9. It is subject to the license
+ * terms in the LICENSE file found in the top-level directory of this
+ * distribution and at http://akaros.cs.berkeley.edu/files/Plan9License. No
+ * part of the UCB release of Plan 9, including this file, may be copied,
+ * modified, propagated, or distributed except according to the terms contained
+ * in the LICENSE file.
+ */
+
+#include "stdio_impl.h"
+#include <unistd.h>
+#include <string.h>
+#include "sys9.h"
+#include "lib.h"
+#include "dir.h"
+
+FILE *
+tmpfile(void){
+	FILE *f;
+	static char name[]="/tmp/tmpfile_XXXXXX";
+	int n;
+
+	snprintf (name, 36, "%x", rand());
```

Comment:
Won't this overwrite /tmp/tmpfile_XXXXX etc. with just a random number like
1341254451
and make the file in the current directory?
Don't you want something like 
snprintf(&name[sizeof("/tmp/tmpfile_"))], 6, "%x", rand());
?

you can even do this:
#define TMPBASE "/tmp/tmpfile_"
#define XXX "XXXXXX"
static char name = TMPBASE+XXX
snprintf(&name[sizeof(TMPBASE)], len(XXX)-1, "%x", rand());